File: frontend/src/app/model/panel.ts
import { RestModel } from "./rest-model";

export class Panel extends RestModel<Panel> {
    title: string;
    url: string;
    headline: string;
    footer: string;
    content: string;
    type: string;

    serialize(): Object {
        return Object.assign(super.serialize(), {
            "title": this.title,
            "url": this.url,
            "headline": this.headline,
            "footer": this.footer,
            "content": this.content,
            "type": this.type
        });
    }

    deserialize(input: any): Panel {
        this._deserialize(input);
        this.title = input.title;
        this.url = input.url;
        this.headline = input.headline;
        this.footer = input.footer;
        this.content = input.content;
        this.type = input.type;
        return this;
    }
}

File: backend/src/main/java/net/weweave/discooze/backend/domain/Panel.java
package net.weweave.discooze.backend.domain;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
public class Panel extends AbstractPersistentObject {
    @NotNull
    @Column(nullable = false)
    private String title;

    @NotNull
    @Column(nullable = false, unique = true)
    private String url;

    @NotNull
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private PanelType type;

    private String headline;
    private String footer;

    @NotNull
    @Column(nullable = false, length = 5000)
    private String content;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public PanelType getType() {
        return type;
    }

    public void setType(PanelType type) {
        this.type = type;
    }

    public String getHeadline() {
        return headline;
    }

    public void setHeadline(String headline) {
        this.headline = headline;
    }

    public String getFooter() {
        return footer;
    }

    public void setFooter(String footer) {
        this.footer = footer;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}
